### ISSUE TITLE
It takes a long time to open

### ISSUE THREAD

**Issue by Niefee** [PowerShell/PowerShell#24710]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I installed it using Scoop. Whether I open it through Windows Terminal or directly open pwsh.exe in the folder, it always takes a long time. The same thing happens when I open it with administrator privileges.

### Expected behavior

```console
quickly open
```

### Actual behavior

```console
Need to wait for a long time.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.6
PSEdition                      Core
GitCommitId                    7.4.6
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![mx_screencap_20241226_085636](https://github.com/user-attachments/assets/14fa3033-3366-4c83-b8bd-e756865357ac)

**Comment by MatejKafka** [comment]
Can you give more specific numbers (how long does it take to start pwsh)?

**Comment by Niefee** [comment]
> Can you give more specific numbers for the startup latency? You can use the following snippet:
> 
> ```powershell
> Measure-Command {pwsh -command "exit 0"}
> ```

Days              : 0
Hours             : 0
Minutes           : 2
Seconds           : 27
Milliseconds      : 312
Ticks             : 1473121682
TotalDays         : 0.00170500194675926
TotalHours        : 0.0409200467222222
TotalMinutes      : 2.45520280333333
TotalSeconds      : 147.3121682
TotalMilliseconds : 147312.1682

**Comment by MatejKafka** [comment]
That's really long, should be under a second. I'd suspect that either your `Documents` folder is on a very slow network drive / OneDrive or you're invoking something very slow in your `$PROFILE`.

How about the following invocation?
```powershell
Measure-Command {pwsh -noprofile -command "exit 0"}
```

**Comment by Niefee** [comment]
> That's really long, should be under a second. I'd suspect that either your `Documents` folder is on a very slow network drive / OneDrive or you're invoking something very slow in your `$PROFILE`.
> 
> How about the following invocation?
> 
> ```powershell
> Measure-Command {pwsh -noprofile -command "exit 0"}
> ```

```
PS C:\> Measure-Command {pwsh -noprofile -command "exit 0"}

Days              : 0
Hours             : 0
Minutes           : 2
Seconds           : 25
Milliseconds      : 136
Ticks             : 1451367701
TotalDays         : 0.00167982372800926
TotalHours        : 0.0403157694722222
TotalMinutes      : 2.41894616833333
TotalSeconds      : 145.1367701
TotalMilliseconds : 145136.7701
```
@MatejKafka

**Comment by MatejKafka** [comment]
Hmm. What about the old Windows PowerShell?
```powershell
Measure-Command {powershell -noprofile -command "exit 0"}
```

**Comment by Niefee** [comment]
> Hmm. What about the old Windows PowerShell?
> 
> ```powershell
> Measure-Command {powershell -noprofile -command "exit 0"}
> ```

```
PS C:\> Measure-Command {powershell -noprofile -command "exit 0"}

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 0
Milliseconds      : 364
Ticks             : 3641102
TotalDays         : 4.21423842592593E-06
TotalHours        : 0.000101141722222222
TotalMinutes      : 0.00606850333333333
TotalSeconds      : 0.3641102
TotalMilliseconds : 364.1102
```
@MatejKafka  This is very fast

**Comment by 237dmitry** [comment]
Try to turm off update checking. Set environment name `POWERSHELL_UPDATECHECK` with value `Off`

**Comment by mklement0** [comment]
Good point, @237dmitry; another potential problem source is telemetry, which can be deactivated by setting environment variable `POWERSHELL_TELEMETRY_OPTOUT` to `1`

**Comment by Niefee** [comment]
@MatejKafka @237dmitry @mklement0 
After setting the environment variables, the opening time is normal:
`POWERSHELL_TELEMETRY_OPTOUT = 1`
`POWERSHELL_UPDATECHECK = Off`

```shell
PowerShell 7.4.6
PS D:\> Measure-Command {pwsh -command "exit 0"}

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 1
Milliseconds      : 419
Ticks             : 14191538
TotalDays         : 1.64253912037037E-05
TotalHours        : 0.000394209388888889
TotalMinutes      : 0.0236525633333333
TotalSeconds      : 1.4191538
TotalMilliseconds : 1419.1538

PS D:\> Measure-Command {pwsh -noprofile -command "exit 0"}

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 1
Milliseconds      : 60
Ticks             : 10603394
TotalDays         : 1.22724467592593E-05
TotalHours        : 0.000294538722222222
TotalMinutes      : 0.0176723233333333
TotalSeconds      : 1.0603394
TotalMilliseconds : 1060.3394

PS D:\> Measure-Command {powershell -noprofile -command "exit 0"}

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 0
Milliseconds      : 382
Ticks             : 3827769
TotalDays         : 4.43028819444444E-06
TotalHours        : 0.000106326916666667
TotalMinutes      : 0.006379615
TotalSeconds      : 0.3827769
TotalMilliseconds : 382.7769
```

**Comment by jvmap** [comment]
I have a similar problem with PowerShell 7.4.6, but also different.

I've set 
```
POWERSHELL_TELEMETRY_OPTOUT = 1
POWERSHELL_UPDATECHECK = Off
```
These should probably be the default, anyway, especially the telemetry one. I never consented to any telemetry collection by PowerShell.

`Measure-Command {pwsh -command "exit 0"}` returns 155 ms, but opening a PowerShell command prompt still takes 15 seconds. This is on Windows 11. Now everything feels sluggish on Windows 11, regardless of the hardware it's given, but this is pretty extreme.

This is the relevant part of the settings .json of the Windows 11 terminal:
```
{
      "guid": "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
      "hidden": false,
      "name": "PowerShell",
      "source": "Windows.Terminal.PowershellCore",
      "startingDirectory": "c:\\code"
  }
```

My Documents folder is on a very fast, local SSD and my `$PROFILE` is empty. During the 15 seconds there is no CPU, disk or network activity that I can notice via taskman, resource manager or `procexp64`.

**Comment by LazyBoot** [comment]
I've noticed that it seems like ever since 7.5.0 released powershell requires all HDDs to spin up when it's starting up. 
With all disks spinning, it only takes around 600-650ms to start, while if one or both my HDDs are parked it'll be more like 6000-8000ms. (Neither powershell, my documents or appdata is on either of those disks, all are on an SSD)
Prior to 7.5.0, I don't ever remember startup times going over 1000ms.
Running on Win11 24H2

**Comment by powercode** [comment]
To give us a reasonable chance to help here, please read https://github.com/PowerShell/PowerShell/tree/master/tools/performance.

Having a .etl file gives us much more to work with.

**Comment by Niefee** [comment]
The problem still exists now, it takes a long time to open. This problem occurs randomly and is now very frequent

@MatejKafka

**Comment by sangafabrice** [comment]
Very much frequent

**Comment by powercode** [comment]
As before, a performance recording would help, otherwise we are just guessing

**Comment by genewitch** [comment]
With the "default" windows console host, powershell 7.5.2 starts fast (n = 2); however with windows terminal preview, it takes 30-60 seconds (n = 30). 

this has been an issue since 7.5 was installed here, but today i noticed that powershell process in "apps" of task manager shows GPU at 100%. 

curious if anyone else has noticed 100% GPU usage in the "Processes > Apps" section of Task manager. the GPU section in "performance" showed nothing in 3d but i could not catch the utilization percentage in there before it started up.

i have cargo culted dragging the window around, clicking in it then out of it, all manner of random rituals to force windows to pay attention to the button i clicked (powershell!)

windows terminal has tabs and ^R lookup, so using console host is out of the question.

**Comment by feg545** [comment]
I use ProcMon.exe found pwsh.exe will try to connect to an http site with 3 times. that site is unaccessable now.
this step waste 12 seconds.

<img width="1603" height="773" alt="Image" src="https://github.com/user-attachments/assets/feaf39e2-e4c7-4226-9b9f-a0887afdb1c6" />

**Comment by feg545** [comment]
Then I use Fiddler capture http trafic, and find this error.

<img width="1460" height="131" alt="Image" src="https://github.com/user-attachments/assets/ede59e3d-7c83-4960-8d3b-1d743f0fd70f" />

http://ctldl.windowsupdate.com/msdownload/update/v3/static/trustedr/en/disallowedcertstl.cab?58c82ec72e89ec74
http://ctldl.windowsupdate.com/msdownload/update/v3/static/trustedr/en/pinrulesstl.cab?91ee4a67d82c51f8

connect this site is timeout
